### PR TITLE
Create appveyor.yml

### PR DESCRIPTION
**Pull request by Dylan-DPC**
Ported some of #200 in a seperate PR so that we can merge this faster and test builds on AppVeyor instead of waiting for #200 to be tested and merged which may take some time. 

Supersedes #225

### CODE REVIEW

**Comment by Dylan-DPC** [comment]
@ceph3us any clue why this config can't find cargo? I just copied it from your fork and it works there

**Comment by ceph3us** [comment]
From a quick glance at appveyor (line 30 [here](https://ci.appveyor.com/project/Dylan-DPC/sodiumoxide/build/0.1.66/job/w3r5ctyo0p8nryaq#L30)) it appears there's line ending issues that's causing a bunch of the build script to end up on a single line (and not work as a result).

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    LLVM_TARGET: win32
+install:
+- cmd: >-
+    curl -sSf -o rustup-init.exe https://win.rustup.rs/
+    
+    curl -sSf http://releases.llvm.org/%LLVM_VERSION%/LLVM-%LLVM_VERSION%-%LLVM_TARGET%.exe -o LLVM.exe
+    
+    7z x LLVM.exe -oC:\LLVM
+    
+    set LIBCLANG_PATH=C:\LLVM\bin
```
Comment (ceph3us) [inline appveyor.yml]: Ah, having looked more carefully this seems to be where the issue lies.

Appveyor treats lines in these blocks as continutions unless there's a blank line between each (as above). This seems to have been lost in the lines below.

**Comment by Dylan-DPC** [comment]
@ceph3us thanks. I have fixed that but I get another error now

**Comment by ceph3us** [comment]
Sorry for any confusions, the double spacing needs to be added, rather than removed, as can be seen in [this commit](https://github.com/sodiumoxide/sodiumoxide/blob/39fad7e780d7e24a8859c98f4959f8b8cca4b446/appveyor.yml).

**Comment by Dylan-DPC** [comment]
Oops. Misread your first comment. Here we go again :joy: :

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    
+    cargo -V
+    
+    vcpkg install libsodium:%VCPKG_TARGET%
+
+cache:
+- C:\Users\appveyor\.cargo\registry
+- target
+build_script:
+- cmd: >-
```
Comment (ceph3us) [inline appveyor.yml]: You'll need to do the same thing here too.

File: appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    cargo build --target %TARGET%
+    cargo build --target %TARGET% --release
+test_script:
+- cmd: >-
```
Comment (ceph3us) [inline appveyor.yml]: And here. That should be it.

File: appveyor.yml (hunk 2 above)
Comment (Dylan-DPC) [reply appveyor.yml]: Thanks

**Comment by Dylan-DPC** [comment]
Appveyor works now but it can't find libsodium which is the next issue to solve.

**Comment by ceph3us** [comment]
This is more-or-less the outline of what's in #200 currently - we'd need to add some alternate way to locate and use libsodium for this in the time being.

**Comment by Dylan-DPC** [comment]
Yeah it is. Will get to fixing it later.

**Comment by 0e4ef622** [comment]
Setting `SODIUM_INC_DIR` to `C:/Tools/vcpkg/packages/libsodium_%VCPKG_TARGET%/include/` and `SODIUM_LIB_DIR` to `C:/Tools/vcpkg/packages/libsodium_%VCPKG_TARGET%/bin/` (or maybe `.../lib/` I'm not sure) should hopefully get the compiler to look in the right place, but `rustc` will still be looking for a `sodium.lib` instead of `libsodium.lib`

**Comment by kpcyrd** [comment]
Just as a notice, travis announced windows builds: https://blog.travis-ci.com/2018-10-11-windows-early-release

**Comment by Dylan-DPC** [comment]
@kpcyrd yes but some people are having issues with windows builds so waiting for things to settle.

**Comment by kurnevsky** [comment]
@Dylan-DPC what the status of this PR? It seems appveyor can't build sodiumoxide because `SODIUM_STATIC` and `SODIUM_INC_DIR` are not defined?

**Comment by Dylan-DPC** [comment]
Superseded by #269

**Pull request closed without merging**